src/wallet/CarbonWallet.ts

```diff
@@ -803,19 +867,21 @@ export class CarbonWallet {
     return error?.message?.includes(`account ${address} not found`);
   };
 
+  private isTxHashNotFound = (error?: Error, hash?: string) => {
+    return error?.message?.includes(`tx (${hash}) not found`);
+  };
+
 
   private isNonceMismatchError = (error?: Error) => {
-    const regex =
```

Comment:
previously this regex extracts the correct account sequence so that it can be used without having to reload account data. why do we want to change that? and if we do want to change this we can remove the logic that assumes return of expected account nonce.